### ISSUE TITLE
Comment inside of a match under-indented

### ISSUE THREAD

**Issue by sfackler** [rust-lang-nursery/rustfmt#680]
I couldn't easily make a small reproduction, but this comment ends up being under-indented when rustfmt is run, interestingly even with the `rustfmt_skip`: https://github.com/sfackler/rust-postgres/blob/4bdbcd82656abd3cfe0d5ea16ddb840f72c95cec/src/lib.rs#L480

**Comment by nrc** [comment]
Did you try this with a very recent rustfmt? I merged a PR this morning that might fix this.

**Comment by sfackler** [comment]
Ah yeah, looks like it's fixed now. Thanks!

**Comment by sfackler** [comment]
Looks like this is back off of a fresh build from git unfortunately.

**Comment by Zarokka** [comment]
seems to be fix now (tested with current master and 0.6.2)

```rust
struct InnerConnection {
}

impl InnerConnection {
    fn setup_typeinfo_query(&mut self) -> result::Result<(), ConnectError> {
        match self.raw_prepare(TYPEINFO_QUERY, "") {
            Ok(..) => return Ok(()),
            Err(Error::IoError(e)) => return Err(ConnectError::IoError(e)),
            // Range types weren't added until Postgres 9.2, so pg_range may not exist
            Err(Error::DbError(ref e)) if e.code() == &SqlState::UndefinedTable => {}
            Err(Error::DbError(e)) => return Err(ConnectError::DbError(e)),
            _ => unreachable!(),
        }
    }
}
```
this snippet stays as is
